viewing_party/party.py

```diff
@@ -1,23 +1,130 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import USER_DATA_2
+
+
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    movie_dict["title"] = title
+    movie_dict["genre"] = genre
+    movie_dict ["rating"] = rating
+    
+    movie_info = [title,genre,rating]
+    if None in movie_info:
+        return None
+    else:
+        return movie_dict
+
+def add_to_watched(user_data,movie):
+    user_data = {}
+    user_data["watched"] = [movie]
+
+    return user_data
+
+def add_to_watchlist(user_data,movie):
+    user_data = {}
+    user_data["watchlist"] = [movie]
+
+    return user_data
+
+def watch_movie(user_data,title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    watched_value = []
+    avg = 0
+    
+    for key in user_data["watched"]:
+        avg += 1
+        watched_value.append(key["rating"])
+    
+    if avg == 0:
+        return 0.0
+    
+    average = sum(watched_value) / len (watched_value)
+    return average
+
+def get_most_watched_genre(user_data):
+    watched_genre = []
+    for key in user_data["watched"]:
+        watched_genre.append(key["genre"])
+        return max(set(watched_genre), key = watched_genre.count)
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    my_movies = []
+    same_movies = []

```

Comment:
Consider renaming same_movies to something more descriptive like "friends_watched_movies"